### PR TITLE
[Neo Core Add] add char support

### PR DESCRIPTION
**Pull request by Jim8y**
# Description

We allows user to use char type in contract, however, user can not use char as the parameter, thus having this pr. 

Fixes # 

## Type of change

- [x] Optimization (the change is only an optimization)
- [ ] Style (the change is only a code style for better maintenance or standard purpose)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

**Test Configuration**:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/Neo/VM/Helper.cs [hunk 1]
```diff
@@ -217,6 +217,9 @@ public static ScriptBuilder EmitPush(this ScriptBuilder builder, object obj)
                 case short data:
                     builder.EmitPush(data);
                     break;
+                case char data:
+                    builder.EmitPush((short)data);
```
Comment (Jim8y) [inline src/Neo/VM/Helper.cs:221]:
```suggestion
                    builder.EmitPush((ushort)data);
```

File: src/Neo/VM/Helper.cs [hunk 2]
```diff
@@ -217,6 +217,9 @@ public static ScriptBuilder EmitPush(this ScriptBuilder builder, object obj)
                 case short data:
                     builder.EmitPush(data);
                     break;
+                case char data:
+                    builder.EmitPush((ushort)data);
```
Comment (cschuchardt88) [inline src/Neo/VM/Helper.cs:221]: we do not allow unicode. so use `byte`

Comment (Jim8y) [reply src/Neo/VM/Helper.cs:221]: ill put it another way, we support char and consider it as ushort. this pr just fix the push method.

Comment (cschuchardt88) [reply src/Neo/VM/Helper.cs:221]: How you doing to convert a `21-bit` UTF-16 to a `char`? That would require an `int32`. At least use `Convert.ToUInt16` or `(ushort)(object)data`

Comment (Jim8y) [reply src/Neo/VM/Helper.cs:221]: sure，will update and add ut

**Comment by Jim8y** [comment]
@cschuchardt88 according to the test, existing code alrady work, do i still need to update?

**Review by shargon — APPROVED** [review]

**Comment by cschuchardt88** [comment]
> @cschuchardt88 according to the test, existing code alrady work, do i still need to update?

Doing it as `(ushort)(object)` is `checked` vs `overflowing`

**Comment by Jim8y** [comment]
> (ushort)(object)

I dont understand, how could char to ushort be overflow?

**Comment by cschuchardt88** [comment]
> > (ushort)(object)
> 
> I dont understand, how could char to ushort be overflow?

`UTF-32` can be 21-bits long for a single character. Like an `emoji` characters.

**Comment by Jim8y** [comment]
@cschuchardt88

**Comment by Jim8y** [comment]
> > > > (ushort)(object)
> > > 
> > > 
> > > I dont understand, how could char to ushort be overflow?
> > 
> > 
> > `UTF-32` can be 21-bits long for a single character. For an example `emoji` character.
> 
> Could you add a UT for this case?

There is no such character that can be assigned to char making char longer than 2 bytes.

**Review by cschuchardt88 — APPROVED** [review]

**Comment by cschuchardt88** [comment]
> > > > (ushort)(object)
> > > 
> > > 
> > > I dont understand, how could char to ushort be overflow?
> > 
> > 
> > `UTF-32` can be 21-bits long for a single character. For an example `emoji` character.
> 
> Could you add a UT for this case?

`char` only supports `ushort` so its ok

**Pull request merged**